static/js/public/details/tabs.js

```diff
@@ -0,0 +1,122 @@
+class Tabs {
+  constructor(el, historyState, children) {
+    this.el = el;
+    this.tabs = {};
+    this.historyState = historyState;
+    this.children = children;
+
+    const tabLinks = this.el.querySelectorAll("[role='tab']");
+
+    tabLinks.forEach((tabLink, i) => {
```

Comment:
perhaps we should check if `tabLinks` exists first? Also, I am not sure if anything has changed since, but in the past I was advised not to use `forEach` directly on DOM nodes but instead do `[].slice.call(this.el.querySelectorAll("[role='tab']"))`.

Reply:
I've added charmhubs first polyfill to cover this